scripts/convert-bson-corpus-tests.php

```diff
@@ -2,6 +2,18 @@
 
 require_once __DIR__ . '/../tests/utils/tools.php';
 
+$expectedFailures = [
+    'Double type: 1.23456789012345677E+18' => 'Variation in double\'s string representation (SPEC-850)',
+    'Double type: -1.23456789012345677E+18' => 'Variation in double\'s string representation (SPEC-850)',
+    'Int64 type: -1' => 'PHP encodes integers as 32-bit if range allows',
+    'Int64 type: 0' => 'PHP encodes integers as 32-bit if range allows',
+    'Int64 type: 1' => 'PHP encodes integers as 32-bit if range allows',
+    'Javascript Code with Scope: bad scope doc (field has bad string length)' => 'Depends on PHPC-889',
+    'Javascript Code with Scope: Unicode and embedded null in code string, empty scope' => 'Embedded null in code string is not supported in libbson (CDRIVER-1879)',
+    'Multiple types within the same document: All BSON types' => 'PHP encodes integers as 32-bit if range allows',
+    'Top-level document validity: Bad $date (number, not string or hash)' => 'Legacy extended JSON $date syntax uses numbers (CDRIVER-2223)',
+];
```

Comment:
This is a welcome improvement inspired by [test-bson-corpus.c](https://github.com/mongodb/libbson/blob/4affa6aa91ce896eca7f1b4326b49bcf1d0c16f5/tests/test-bson-corpus.c#L19) in libbson. No longer will we need to manually insert (and restore) `--XFAIL--` blocks in generated test files.